src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -238,70 +227,90 @@
       Overwrite="true" />
   </Target>
 
-  <!-- #################################### -->
-  <!-- Generate BuildVersion.props -->
-  <!-- #################################### -->
-  <Target Name="CreateOrUpdateCurrentVersionFile" Condition="'$(SkipVersionGeneration)'!='true'" DependsOnTargets="GenerateCurrentVersion;GenerateBuiltByString">
-    <!-- Error out if GeneratedBuildNumberMajor is empty since GenerateCurrentVersion should have calculated it already. -->
-    <Error Condition="'$(GeneratedBuildNumberMajor)'==''" Text="Error when generating current version." />
-
+  <!-- 
+    Target: CreateOrUpdateCurrentVersionFile
+
+    Generates the version data and saves it to $(BuildVersionFile) if it hasn't been imported
+
+    Outputs:
+      $(LatestCommit)
+      $(BuiltByString)
+      $(BuildNumberMajor)
+      $(BuildNumberMinor)
+      $(AssemblyFileVersion)
+  -->
+  <Target Name="CreateOrUpdateCurrentVersionFile"
+      BeforeTargets="ResolveProjectReferences"
+      Condition="'$(SkipVersionGeneration)' != 'true' AND '$(VersionPropsImported)' != 'true'">
```

Comment:
If I understand correctly this condition: `'$(VersionPropsImported)' != 'true'` is the case where the file exists.  If that is true, when will you ever "Update" the Version file?